roles/roles.tf

```
@@ -774,20 +800,197 @@ data "aws_iam_policy_document" "fargate_logging_policy" {
+    GLOBAL_DYNAMO_TBLES     = jsonencode(local.global_dynamo_tables)
+  }
+}
+
+# EKS [Common : Databricks and EMR]
+resource "aws_iam_policy" "satellite_ca_policy" {
+  count = length(var.satellite_regions) > 0 ? 1 : 0
+  name  = "tecton-satellite-ca-policy"
+  policy = data.template_file.satellite_ca_policy_json[0].rendered
+  tags = local.tags
+}
+
+# EKS [Common : Databricks and EMR]
+resource "aws_iam_role_policy_attachment" "satellite_ca_policy_attachment" {
+  count      = length(var.satellite_regions) > 0 ? 1 : 0
+  policy_arn = aws_iam_policy.satellite_ca_policy[0].arn
+  role       = aws_iam_role.eks_node_role.name
+}
+
+# EKS [Common : Databricks and EMR]
+resource "aws_iam_policy" "cross_account_satellite_region_policy" {
+  count = length(var.satellite_regions) > 0 ? 1 : 0
+  name  = "tecton-${var.deployment_name}-cross-account-satellite-region-policy"
+  policy = file("${path.module}/../templates/satellite_serving_dynamodb_policy.json")
+  tags = local.tags
+}
+
+# EKS [Common : Databricks and EMR]
+resource "aws_iam_role_policy_attachment" "cross_account_satellite_region_policy_attachment" {
+  count      = length(var.satellite_regions) > 0 ? 1 : 0
+  policy_arn = aws_iam_policy.cross_account_satellite_region_policy[0].arn
+  role       = aws_iam_role.eks_node_role.name
+}
+
+# DEVOPS [Common : Databricks and EMR]
+data "template_file" "satellite_devops_policy_json" {
+  count    = length(var.satellite_regions) > 0 ? 1 : 0
+  template = file("${path.module}/../templates/satellite_devops_policy.json")
+  vars = {
+    ACCOUNT_ID      = var.account_id
+    DEPLOYMENT_NAME = var.deployment_name
+    SECURITY_GROUPS = jsonencode(local.security_groups)
+  }
+}
+
+# DEVOPS [Common: Databricks and EMR]
+resource "aws_iam_policy" "satellite_devops_policy" {
+  count  = length(var.satellite_regions) > 0 ? 1 : 0
+  name   = "tecton-${var.deployment_name}-satellite-devops-policy"
+  policy = data.template_file.satellite_devops_policy_json[0].rendered
+  tags   = local.tags
+}
+
+# DEVOPS [Common: Databricks and EMR]
+resource "aws_iam_role_policy_attachment" "satellite_devops_policy_attachment" {
+  count      = length(var.satellite_regions) > 0? 1 : 0
+  policy_arn = aws_iam_policy.satellite_devops_policy[0].arn
+  role       = aws_iam_role.devops_role.name
+}
+
+# Fargate satellite [Common : Databricks and EMR]
+data "template_file" "eks_satellite_fargate_node" {
+  for_each = toset(var.satellite_regions)
+
+  template = file("${path.module}/../templates/fargate_eks_role.json")
+  vars = {
+    ACCOUNT_ID          = var.account_id
+    ASSUMING_ACCOUNT_ID = var.tecton_assuming_account_id
+    DEPLOYMENT_NAME     = var.deployment_name
+    REGION              = each.value
+  }
+}
+
+# Fargate satellite [Common : Databricks and EMR]
+resource "aws_iam_policy" "eks_fargate_satellite_node_policy" {
+  for_each = toset(var.satellite_regions)
+
+  name   = "tecton-${var.deployment_name}-${each.key}-eks-fargate-node-policy"
```

This name is likely to hit the character limit.
```suggestion
  name   = "tecton-${var.deployment_name}-${each.key}-eks-fargate"
```